docs/src/modules/spring/pages/workflow-entities.adoc

```diff
@@ -0,0 +1,100 @@
+= Implementing Workflow Entities
+:page-aliases: spring:workflow.adoc
+:sample-url: https://github.com/lightbend/kalix-jvm-sdk/tree/main/samples/spring-transfer-workflow
+
+//TODO prepare opening paragraph
+
+== Modeling workflow state
+
+Workflow state design should be oriented around data that are not required to run all workflow step but also to track workflow progress.
```

Comment:
I don't think we should be prescriptive about what should be put on the Workflow state. I mean, we don't need to say something like 'it should or it should not'. 

Better if we say that a Workflow has a state and that on each iteration we can update it. We can use it to track progress or to validate steps. Or whatever we see fitting.